mitm/mitm.go

```diff
@@ -51,6 +52,7 @@ type Config struct {
 	getCertificate func(*tls.ClientHelloInfo) (*tls.Certificate, error)
 	roots          *x509.CertPool
 	skipVerify     bool
+	TLSHandshakeErrorHandler func(*http.Request, error)
```

Comment:
TLSHandshakeErrorHandler -> TLSHandshakeErrorCallback: done

If I don't export it then I need a function on mitm.Config to call the function for me since mitm is it's own package. Still think it makes sense?

Currently the martian package calls p.mitm.TLSHandshakeErrorCallback() directly.